### PR TITLE
[WIP] Use lazy inputs and bigger partition for ISS & FOM

### PR DESCRIPTION
**Pull request by schillic**
* [x] Add high-dimensional check settings.
* [x] Use bigger partition for ISS and FOM.
* [x] Add setting for ISS in dense mode.
* [x] Use lazy inputs for ISS and FOM.
* [x] Use lazy exponentiation for FOM's discretization.
* Not realistic: Find the right setting for FOM in dense mode.

### CODE REVIEW

File: models/SLICOT/fom/fom.jl [hunk 1]
```diff
@@ -29,18 +29,20 @@ function fom(input_options::Options)
...
+                                  :partition => partition_2D,
                                   :plot_vars => [0, 1],
                                   :assume_sparse => true)
         # :projection_matrix => sparse(read(matopen(@relpath "out.mat"), "M")),
     elseif input_options[:mode] == "check"
         problem_options = Options(:vars => 1:1006,
-                                  :partition => partition,
+                                  :partition => partition_6D,
                                   :property => property,
+                                  :lazy_inputs_interval => 2147483647,
```
Comment (mforets) [inline models/SLICOT/fom/fom.jl]: shouldn't we allow something like `:lazy_inputs_interval => -1` ?

**Comment by mforets** [comment]
there's something i don't understand:

```julia
# discrete time
sol = solve(problem_TV, :approx_model=>"nobloating", :T=>time_horizon, :δ=>5e-3, :vars=>136:270,
                        :mode=>"check", :property=>y3SPEC2, :projection_matrix=>C, :assume_sparse=>true,
                        :lazy_inputs_interval=>100,
                        :partition=>[1:135, 136:270]);
sol.satisfied, sol.violation

AssertionError: expected CacheMinkowskiSum type, got LazySets.Hyperrectangle{Float64}

Stacktrace:
 [1] (::Reachability.Properties.#_f#13)(::Int64, ::Int64, ::LazySets.MinkowskiSum{Float64,LazySets.Hyperrectangle{Float64},LazySets.LinearMap{Float64,Float64}}) at /Users/forets/.julia/v0.6/Reachability/src/Properties/check_property.jl:154
 [2] check_blocks(::SparseMatrixCSC{Float64,Int64}, ::Array{LazySets.LazySet{Float64},1}, ::MathematicalSystems.ConstantInput{LazySets.LinearMap{Float64,Float64}}, ::Reachability.Properties.#_f#13, ::Int64, ::Int64, ::Array{Int64,1}, ::Array{UnitRange{Int64},1}, ::Bool, ::Reachability.Properties.LinearConstraintProperty{Float64}) at /Users/forets/.julia/v0.6/Reachability/src/Properties/check_blocks.jl:104
 [3] #check_property#2(::String, ::Float64, ::Type{LazySets.Hyperrectangle}, ::Float64, ::Type{LazySets.Hyperrectangle}, ::Bool, ::Bool, ::Bool, ::Array{Any,1}, ::Reachability.Properties.#check_property, ::MathematicalSystems.InitialValueProblem{MathematicalSystems.ConstrainedLinearControlDiscreteSystem{Float64,SparseMatrixCSC{Float64,Int64},Void,MathematicalSystems.ConstantInput{LazySets.LinearMap{Float64,Float64}}},LazySets.BallInf{Float64}}, ::Int64) at /Users/forets/.julia/v0.6/Reachability/src/Properties/check_property.jl:189
 [4] (::Reachability.Properties.#kw##check_property)(::Array{Any,1}, ::Reachability.Properties.#check_property, ::MathematicalSystems.InitialValueProblem{MathematicalSystems.ConstrainedLinearControlDiscreteSystem{Float64,SparseMatrixCSC{Float64,Int64},Void,MathematicalSystems.ConstantInput{LazySets.LinearMap{Float64,Float64}}},LazySets.BallInf{Float64}}, ::Int64) at ./<missing>:0
 [5] solve(::MathematicalSystems.InitialValueProblem{MathematicalSystems.ConstrainedLinearControlContinuousSystem{Float64,SparseMatrixCSC{Float64,Int64},Void,MathematicalSystems.ConstantInput{LazySets.LinearMap{Float64,Float64}}},LazySets.BallInf{Float64}}, ::Reachability.Options) at /Users/forets/.julia/v0.6/Reachability/src/solve.jl:200
 [6] solve(::MathematicalSystems.InitialValueProblem{MathematicalSystems.ConstrainedLinearControlContinuousSystem{Float64,SparseMatrixCSC{Float64,Int64},Void,MathematicalSystems.ConstantInput{LazySets.LinearMap{Float64,Float64}}},LazySets.BallInf{Float64}}, ::Pair{Symbol,String}, ::Pair{Symbol,Float64}, ::Vararg{Pair{Symbol,#s65} where #s65,N} where N) at /Users/forets/.julia/v0.6/Reachability/src/solve.jl:236
```

BUT

```julia
# discrete time
sol = solve(problem_TV, :approx_model=>"nobloating", :T=>time_horizon, :δ=>5e-3, :vars=>136:270,
                        :mode=>"check", :property=>y3SPEC2, :projection_matrix=>C, :assume_sparse=>true,
                        :lazy_inputs_interval=>1500,
                        :partition=>[1:135, 136:270]);
sol.satisfied, sol.violation
```

and finally

```julia
# discrete time
sol = solve(problem_TV, :approx_model=>"nobloating", :T=>time_horizon, :δ=>5e-3, :vars=>136:270,
                        :mode=>"check", :property=>y3SPEC2, :projection_matrix=>C, :assume_sparse=>true,
                        :lazy_inputs_interval=>4000,
                        :partition=>[1:135, 136:270]);
sol.satisfied, sol.violation
```

why do we get an `AssertionError` for the first one?

File: models/SLICOT/fom/fom.jl (hunk 1 above)
Comment (schillic) [reply models/SLICOT/fom/fom.jl]: Good idea, I created https://github.com/JuliaReach/Reachability.jl/issues/201 :+1:

**Comment by schillic** [comment]
> why do we get an AssertionError for the first one?

Good finding, I have no idea... I will check.

**Comment by mforets** [comment]
FWIW, those errors in [this comment](https://github.com/JuliaReach/ReachabilityBenchmarks/pull/24#issuecomment-391130234) are now fixed 👌

**Comment by schillic** [comment]
Merge?

**Pull request merged**